pkg/cloud/services/networking/router.go

```diff
@@ -165,6 +165,34 @@ INTERFACE_LOOP:
 	return nil
 }
 
+func (s *Service) DeleteRouter(network *infrav1.Network) error {
+	if network.Router == nil || network.Router.ID == "" {
+		s.logger.V(4).Info("No need to delete router since no router exists.")
+		return nil
+	}
+	exists, err := s.existsRouter(network.Router.ID)
+	if err != nil {
+		return err
+	}
+	if !exists {
+		s.logger.Info("No router", "router", network.Router.ID)
+		return nil
+	}
+	if network.Subnet == nil || network.Subnet.ID == "" {
+		s.logger.V(4).Info("No need to remove router interface since no subnet exists.")
```

Comment:
Why do we skip the whole router deletion if subnet doesn't exist? I think we only have to skip `RemoveInterface` ?

Reply:
@sbueringer Right, I will fix.